src/features/home/components/BusinessModule.tsx

```diff
@@ -1,21 +1,50 @@
-import React from 'react'
+import { t } from '@lingui/macro'
+import React, { useEffect, useState } from 'react'
 import { Dimensions, PixelRatio, Linking } from 'react-native'
 import styled from 'styled-components/native'
 
+import { useUserProfileInfo } from 'features/home/api'
 import { IdeaIcon } from 'features/home/assets/IdeaIcon'
 import { NextArrowIcon } from 'features/home/assets/NextArrowIcon'
 import { BusinessPane } from 'features/home/contentful'
+import { _ } from 'libs/i18n'
+import { useSnackBarContext } from 'ui/components/snackBar/SnackBarContext'
 import { Typo, ColorsEnum, getSpacing, MARGIN_DP, LENGTH_S, RATIO_BUSINESS, Spacer } from 'ui/theme'
 import { BorderRadiusEnum } from 'ui/theme/grid'
 
+import { fillUrlEmail, shouldUrlBeFilled } from './BusinessModule.utils'
+
 export const BusinessModule = ({ firstLine, secondLine, leftIcon, image, url }: BusinessPane) => {
-  const openUrl = () => {
-    url && Linking.openURL(url)
+  const [shouldRedirect, setShouldRedirect] = useState(false)
+
+  const { data: profileData, isLoading: isProfileDataLoading } = useUserProfileInfo()
+  const { displayInfosSnackBar } = useSnackBarContext()
```

Comment:
tu penses vraiment que la snackbar est nécessaire? Si on met un peu trop de temps pour aller chercher l'email, qu'on affiche la snackbar et que 0.5s plus tard on a l'email, ça risque de flicker.

Je me demande si ça vaut pas le coup de modifier AuthContext => UserInfoContext qui donnerait l'email, mais ça risque de faire bcp pour cette PR. T'en penses quoi?

Reply:
Bah je sais que @blixit travaille sur un sujet de "stocker les info user" mais l'idée c'est que même si on les stocks, il faut un temps pour les obtenir.

Et pendant ce temps (normalement très court) on peut pas rediriger
Sauf que
-> en cas d'échec de requête (et donc elle se réitère via react query), ça peut prendre plusieurs secondes.

On a vu avec Mathieu et ça semblait lui aller. 

Maintenant c'est vraiment un edge case quoi